### DISCUSSION TITLE
is a good idea to define a proxy within react component?

### DISCUSSION THREAD

**Discussion by lamualfa** [pmndrs/valtio#417] [Q&A] [answered]
## code

```js
const $state = useConstant(() => proxy())
const state$ = useSnapshot($state)
```

- use `$state` to manipulate the state value
- use `state$` to read the state value

**Comment by dai-shi** [comment, marked as answer]
Yes. You can pass `$state` with React Context too.

**Reply by lamualfa** [reply]
How about passing it via `props` to children component instead of react context?

**Reply by dai-shi** [reply]
That's also recommended.
